metadata/utils.go

```diff
@@ -52,19 +52,24 @@ const (
 // TryGetTTL tries to get the ttl as a time.Duration value for pubsub, binding and any other building block.
 func TryGetTTL(props map[string]string) (time.Duration, bool, error) {
 	if val, ok := props[TTLMetadataKey]; ok && val != "" {
-		valInt64, err := strconv.ParseInt(val, 10, 64)
+		duration, err := time.ParseDuration(val)
```

Comment:
We should have two different TTL attributes then. One "ttlInSeconds" and another one as "ttl" - this way, one is an integer and the other one is a duration. Components should agree and not mix them. Accepting unit in "ttlInSeconds"is not correct and the component should be changed. If it is stable, then it needs to follow a deprecation path.

Reply:
We should add `ttl` as an alias instead @artursouza. But that will also require an update to runtime.

Reply:
We were all writing the same thing.

I agree with @berndverst on aliasing `ttl` = `ttlInSeconds`. It's ok if `ttlInSeconds` accepts units too, it's not a breaking change.

Reply:
It is odd because you don't know the granularity of the client underneath. "ms" might not be supported, for example.